nixos/modules/config/nix-channel.nix

```
@@ -18,6 +18,17 @@ let
 
   cfg = config.nix;
 
+  channelsFilePlaintext = pkgs.writeTextFile {
+    name = "plaintext-channels-file";
+    text = builtins.concatStringsSep "\n" config.system.defaultChannels;
+  };
+
+  channelsFileBase64 = pkgs.runCommand "base64-encode-file" { buildInputs = [ pkgs.basez ]; } ''
+    base64 --input="${channelsFilePlaintext}" --output="$out"
```

since it's an executable, it must be in `nativeBuildInputs`
```suggestion
  channelsFileBase64 = pkgs.runCommand "base64-encode-file" { nativeBuildInputs = [ pkgs.basez ]; } ''
    base64 --input="${channelsFilePlaintext}" --output="$out"
```
alternatively, you can use interpolation and that's it
```suggestion
  channelsFileBase64 = pkgs.runCommand "base64-encode-file" { } ''
    ${coreutils}/bin/base64 "${channelsFilePlaintext}" >"$out"
```
btw, why using `basez`, if 1) its upstream is dead and 2) coreutils already provides an executable (note difference in flags: https://man.archlinux.org/man/base64.1)

---

i actually had ```${coreutils}/bin/base64 "${channelsFilePlaintext}" > "$out"``` initially but the final `.nix-channels` file always ended up malformed, containing only part of the expected data, like it was written to concurrently or something.

switching to `basez` made the issue go away. this behavior may be due to a different issue described in https://github.com/NixOS/nixpkgs/pull/464800#discussion_r2559438955

---

let me test this again on my device.